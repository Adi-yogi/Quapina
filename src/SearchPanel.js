import React, { useState } from 'react';
import { connect } from 'react-redux';
import { fetchRestaruants } from 'store/actions.js';

export const SearchPanel = ({ dispatch }) => {
    const [searchInCity, setCity] = useState('');
    const [filterBy, setFilterBy] = useState('');

    const searchRestaurants = event => {
        event.preventDefault();
        dispatch(fetchRestaruants({ searchInCity, filterBy }));
    }
    return <div className="searchForm">
        <form onSubmit={searchRestaurants}>
            <input required="required" type="text" name="searchInCity" placeholder="City Name"
                onChange={event => setCity(event.target.value)}
            />
            <input type="text" name="filterBy" placeholder="Refine"
                onChange={event => setFilterBy(event.target.value)} />
            <button type="submit">Search</button>
        </form>
    </div>
}

export default connect()(SearchPanel);