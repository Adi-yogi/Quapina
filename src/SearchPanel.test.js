import React from 'react';
import { shallow } from 'enzyme';
import { SearchPanel } from './SearchPanel.js';


it('renders without crashing', () => {
    const searchPanel = shallow(<SearchPanel />);
    expect(searchPanel).toMatchSnapshot();
});

it('searchRestaurants on click', () => {
    const dispatch = jest.fn();
    const preventDefault = jest.fn();

    const searchPanel = shallow(<SearchPanel dispatch={dispatch} />);
    searchPanel.find('form').simulate('submit', { preventDefault })
    expect(dispatch.mock.calls.length).toBe(1)
});
