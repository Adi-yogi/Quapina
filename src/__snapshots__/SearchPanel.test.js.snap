// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="searchForm"
>
  <form
    onSubmit={[Function]}
  >
    <input
      name="searchInCity"
      onChange={[Function]}
      placeholder="City Name"
      required="required"
      type="text"
    />
    <input
      name="filterBy"
      onChange={[Function]}
      placeholder="Refine"
      type="text"
    />
    <button
      type="submit"
    >
      Search
    </button>
  </form>
</div>
`;
